crates/rspack_plugin_javascript/src/parser_plugin/import_meta_plugin.rs

```diff
@@ -21,10 +22,32 @@ pub struct ImportMetaPlugin;
 impl ImportMetaPlugin {
   fn import_meta_url(&self, parser: &JavascriptParser) -> String {
     Url::from_file_path(&parser.resource_data.resource)
-      .expect("should be a path")
+      .expect("should be a url")
       .to_string()
   }
 
+  // This is the same as the url.fileURLToPath() of the import.meta.url
```

Comment:
Using `PathBuf::from` directly might be more efficient. However, considering that the `Resource` might contain a query string – which can introduce various edge cases like `./x.js?foo=/../y.js` – processing it through a URL parser is safer.

https://github.com/web-infra-dev/rspack/blob/561581ef0ed9308efa42221572a47ec1d9a57ada/crates/rspack_loader_runner/src/content.rs#L114-L115

- https://github.com/vitejs/vite/issues/19406